tests/by-util/test_split.rs

```diff
@@ -156,3 +161,21 @@ fn test_split_additional_suffix() {
     assert_eq!(glob.count(), 2);
     assert_eq!(glob.collate(), at.read(name).into_bytes());
 }
+
+#[test]
+#[cfg(unix)]
+fn test_filter() {
+    // like `test_split_default()` but run a command before writing
+    let (at, mut ucmd) = at_and_ucmd!();
+    let name = "filtered";
+    let glob = Glob::new(&at, ".", r"x[[:alpha:]][[:alpha:]]$");
+    let n_lines = 3;
+    RandomFile::new(&at, name).add_lines(n_lines);
+
+    // change all characters to 'i'
+    ucmd.args(&["--filter=sed s/./i/g > $FILE", name])
+        .succeeds();
+    // assert all characters are 'i' (command succeded)
+    assert!(glob.collate().iter().find(|&&c| c == ('i' as u8)));
```

Comment:
please fix this error !

Reply:
fixed, although I feel it's not idiomatic the way it turned out